share/qtvcp/screens/qtplasmac/qtplasmac_handler.py

```diff
@@ -493,42 +501,60 @@ def mdi_line_patch(self):
         self.old_submit = MDI_LINE.submit
         MDI_LINE.submit = self.new_submit
 
-    # don't allow M3 or M5 in MDI codes
     def new_submit(self):
+        self.mdiError = False
         intext = str(self.w.mdihistory.MDILine.text()).rstrip()
-        if intext == '': return
-        if intext.upper() == 'HALMETER':
+        if intext == '':
+            self.mdi_show_clicked()
+            return
+        if intext.lower() == 'halmeter':
             AUX_PRGM.load_halmeter()
-        elif intext.upper() == 'STATUS':
+        elif intext.lower() == 'status':
             AUX_PRGM.load_status()
-        elif intext.upper() == 'HALSHOW':
+        elif intext.lower() == 'halshow':
             AUX_PRGM.load_halshow()
-        elif intext.upper() == 'CLASSICLADDER':
+        elif intext.lower() == 'classicladder':
             AUX_PRGM.load_ladder()
-        elif intext.upper() == 'HALSCOPE':
+        elif intext.lower() == 'halscope':
             AUX_PRGM.load_halscope()
-        elif intext.upper() == 'CALIBRATION':
+        elif intext.lower() == 'calibration':
             AUX_PRGM.load_calibration()
-        elif intext.upper() == 'PREFERENCE':
+        elif intext.lower() == 'preference':
             self.new_openReturn(os.path.join(self.PATHS.CONFIGPATH, 'qtplasmac.prefs'))
             self.w.gcode_editor.readOnlyMode()
             self.w.preview_stack.setCurrentIndex(2)
+        elif intext.lower() == 'clear history':
```

Comment:
Might I suggest some of these changes would be a nice addition to the base widget. 
clear history, setp etc sounds like great ideas for the MDI basewidget.
While using uppercase was a choice I made to make sure people didn't accidentally do something they didn't want - I am not against changes -especially if you show that in the real world it works fine.

Please feel free to modify basewidgets if the use case is general enough. I'd be happy to give my opinion on a change if you think you need it, but I think you guys have a good 'feel' for  the idea of Qtvcp's infrastructure .
great ideas here!

Reply:
> Might I suggest some of these changes would be a nice addition to the base widget.
clear history, setp etc sounds like great ideas for the MDI basewidget.

Actually, that's a great idea.  Phill patched the MDI_LINE function a while back because M3, M4, and M5 commands in the MDI cause QtPlasmaC angst.  Due to tunnel vision while working in the handler, I admit I didn't think about pushing this back to the base widget as well.  Sorry about that.

>  While using uppercase was a choice I made to make sure people didn't accidentally do something they didn't want - I am not against changes -especially if you show that in the real world it works fine.

The change from intext.upper() to intext.lower() was just for consistency because our handler primarily compares strings in lower case.  I found myself writing "if" statements that used both upper() and lower() and that just seemed "wrong". 

I'd be honored to add these to the base widget, I will work on a PR in the next few days to add these features, and try harder to remember in the future :)